File: java/033_resistor-color-duo/ResistorColorDuo.java
import java.util.Arrays;

class ResistorColorDuo {
    int value(String[] colors) {
        return colorCode(colors[0]) * 10 + colorCode(colors[1]);
    }
    
    int colorCode(String color) {
        return Arrays.asList(colors()).indexOf(color);
    }
    
    String[] colors() {
        return "black,brown,red,orange,yellow,green,blue,violet,grey,white".split(",");
    }
}

File: csharp/075_resistor-color-duo/ResistorColorDuo.cs
using System;

public static class ResistorColorDuo
{
    public static int Value(string[] colors)
        => ColorCode(colors[0]) * 10 + ColorCode(colors[1]);
    
    static int ColorCode(string color)
        => Array.IndexOf(Colors(), color);
    
    static string[] Colors()
        => "black,brown,red,orange,yellow,green,blue,violet,grey,white".Split(',');
}